javalib/src/main/scala/java/util/HashMap.scala

```diff
@@ -0,0 +1,187 @@
+    this(initialCapacity, HashMap.DEFAULT_LOAD_FACTOR)
+
+  def this() =
+    this(HashMap.DEFAULT_INITIAL_CAPACITY)
+
+  def this(m: ju.Map[_ <: K, _ <: V]) = {
+    this()
+    initFrom(m)
+  }
+
+  protected def initFrom(m: ju.Map[_ <: K, _ <: V]): Unit = {
+    for (entry <- m.entrySet)
+      inner(Box(entry.getKey)) = entry.getValue
+  }
+
+  override def clear(): Unit =
+    inner.clear()
+
+  override def clone(): ju.HashMap[K, V] = {
+    new ju.HashMap(inner.clone(), HashMap.DEFAULT_INITIAL_CAPACITY, HashMap.DEFAULT_LOAD_FACTOR)
+  }
+
+  override def containsKey(key: Any): Boolean =
+    inner.contains(Box(key.asInstanceOf[K]))
+
+  override def containsValue(value: Any): Boolean =
+    inner.valuesIterator.contains(value.asInstanceOf[V])
+
+  override def entrySet(): ju.Set[ju.Map.Entry[K, V]] =
+    new EntrySet
+
+  override def get(key: Any): V =
+    inner.get(Box(key.asInstanceOf[K])).getOrElse(null.asInstanceOf[V])
+
+  override def isEmpty(): Boolean =
+    inner.isEmpty
+
+  override def keySet(): Set[K] =
+    new KeySet
+
+  override def put(key: K, value: V): V = {
+    inner.put(Box(key), value) match {
+      case Some(oldValue) => oldValue
+      case None           => null.asInstanceOf[V]
+    }
+  }
+
+  override def remove(key: Any): V = {
+    val boxedKey = Box(key.asInstanceOf[K])
+    inner.get(boxedKey) match {
+      case Some(value) =>
+        inner -= boxedKey
+        value
+      case None =>
+        null.asInstanceOf[V]
+    }
+  }
+
+  override def size(): Int =
+    inner.size
+
+  override def values(): Collection[V] =
+    new ValuesView
+
+  private class EntrySet extends ju.AbstractSet[ju.Map.Entry[K, V]] with AbstractMapView[ju.Map.Entry[K, V]] {
+    /*
+    Returns a Set view of the mappings contained in this map. The set is backed
+    by the map, so changes to the map are reflected in the set, and vice-versa.
+     If the map is modified while an iteration over the set is in progress (except through
+     the iterator's own remove operation, or through the setValue operation on a map entry
+     returned by the iterator) the results of the iteration are undefined. The set supports
+      element removal, which removes the corresponding mapping from the map, via the
+      Iterator.remove, Set.remove, removeAll, retainAll and clear operations. It
+      does not support the add or addAll operations.
+     */
+
+    override def iterator(): ju.Iterator[ju.Map.Entry[K, V]] = {
+      new AbstractMapViewIterator[ju.Map.Entry[K, V]] {
+        override protected def getNextForm(key: Box[K]): ju.Map.Entry[K, V] =
+          new ju.AbstractMap.SimpleEntry(key.inner, inner(key)) {
+            override def setValue(value: V): V = {
+              inner.update(key, value)
+              super.setValue(value)
+            }
+          }
+      }
+    }
+  }
+
+  private class KeySet extends ju.AbstractSet[K] with AbstractMapView[K] {
+
+    override def remove(o: Any): Boolean = {
```

Comment:
The blank lines at the beginning and end of this class don't help, given the size of the class. I would remove them. Same for `ValuesView` and `AbstractMapView`.